include/oneapi/dpl/pstl/utils.h

```diff
@@ -276,16 +263,32 @@ class __not_equal_value
 template <typename _Pred>
 class __transform_functor
 {
-    _Pred _M_pred;
+    mutable _Pred _M_pred;
 
   public:
-    explicit __transform_functor(_Pred __pred) : _M_pred(__pred) {}
+    explicit __transform_functor(_Pred __pred) : _M_pred(std::move(__pred)) {}
 
     template <typename _Input1Type, typename _Input2Type, typename _OutputType>
     void
-    operator()(const _Input1Type& x, const _Input2Type& y, _OutputType& z) const
+    operator()(const _Input1Type& x, const _Input2Type& y, _OutputType&& output) const
+    {
+        __transform_impl(::std::forward<_OutputType>(output), x, y);
+    }
+
+    template <typename _Input1Type, typename _OutputType>
+    void
+    operator()(_Input1Type&& x, _OutputType&& output) const
+    {
+        __transform_impl(::std::forward<_OutputType>(output), x);
+    }
+
+private:
+    template <typename _OutputType, typename... _Args>
+    void __transform_impl(_OutputType&& output, _Args&&... args) const
     {
-        z = _M_pred(x, y);
+        static_assert(sizeof...(_Args) < 3, "A predicate supports either unary or binary transformation");
```

Comment:
Is this `static_assert` necessary? This is a private method that is only called within this class with either exactly 1 or 2 arguments.